drivers/mge-hid.c

```diff
@@ -300,28 +336,60 @@ static info_lkp_t eaton_abm_status_info[] = {
 	{ 0, NULL, NULL, NULL }
 };
 
+static info_lkp_t eaton_charger_type_info[] = {
+	{ 0, "None", NULL, NULL },
+	{ 1, "Extended (CLA)", NULL, NULL },
+	{ 2, "Large extension", NULL, NULL },
+	{ 3, "Extra large extension (XL)", NULL, NULL },
+	{ 4, "ABM", NULL, NULL },
+	{ 5, "Constant Charge (CC)", NULL, NULL },
+	{ 0, NULL, NULL, NULL }
+};
+
 /* Used to process ABM flags, for ups.status (CHRG/DISCHRG/RB) */
 static const char *eaton_abm_chrg_dischrg_fun(double value)
 {
 	/* Don't process if ABM is disabled */
 	if (advanced_battery_monitoring == ABM_DISABLED)
 		return NULL;
 
-	switch ((long)value)
+	if (dstate_getinfo("battery.charger.type"))
```

Comment:
Same comment as above, why base this on variable existence and not `advanced_battery_monitoring == ABM_ENABLED_TYPE`? Would this not send a device down the wrong path if it does not expose the variable yet (perhaps temporarily not)?